packages/contrast-colors/index.js

```diff
@@ -13,7 +13,480 @@ governing permissions and limitations under the License.
 const d3 = require('./d3.js');
 
 const { catmullRom2bezier, prepareCurve } = require('./curve.js');
-const { color } = require('./d3.js');
+// const { color } = require('./d3.js');
+
+class Theme {
+  constructor({colors, backgroundColor, lightness, contrast = 1, output = 'HEX'}) {
+    this._output = output;
+    this._colors = colors;
+    this._lightness = lightness;
+
+    this._setBackgroundColor(backgroundColor);
+    this._setBackgroundColorValue();
+    
+    this._contrast = contrast;
+    if (!this._colors) {
+      throw new Error(`No colors are defined`);
+    }
+    if (!this._backgroundColor) {
+      throw new Error(`Background color is undefined`);
+    }
+    colors.forEach(color => {
+      if(!color.ratios) throw new Error(`Color ${color.name}'s ratios are undefined`);
```

Comment:
```suggestion
      if (!color.ratios) throw new Error(`Color ${color.name}'s ratios are undefined`);
```
nit: spacing. Could use some linting/prettier action here to auto-clean this stuff up.